pymc3/distributions/bound.py

```diff
@@ -232,32 +133,70 @@ class Bound:
     .. code-block:: python
 
         with pm.Model():
-            NegativeNormal = pm.Bound(pm.Normal, upper=0.0)
```

Comment:
Do we not support this use-case anymore?

Reply:
No, you have to pass a .dist(). There is a very informative error message with a code example if users do that.

This was done to simplify the implementation and also be more in line with the rest of the API (e.g when you specify distributions for Mixtures, LKJCorr priors, RW?). Also in the future, Truncated and Censored distributions would work the same way.

I checked the original PRs for Bound and this was actually the initial intended API, but they couldn't make it work.